src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -316,12 +320,9 @@ public static void AddAlgorithm(Type algorithm, params string[] names)
             }
 
             // Everything looks valid, so we're safe to take the table lock and add the name mappings.
-            lock (s_InternalSyncObject)
+            foreach (string name in algorithmNames)
             {
-                foreach (string name in algorithmNames)
-                {
-                    appNameHT[name] = algorithm;
-                }
+                appNameHT[name] = algorithm;
```

Comment:
lock inside of the loop is I guess fine since this is called relatively rarely and presumably once per process

Reply:
This method should not be called very often, but someone can write the code to call it on every request. 
The reason I removed lock here is because I really needed to remove lock in the CreateFromName method, which causes  contention. See line 336.